### PR TITLE
Safety profile conformance test changes

### PR DESCRIPTION
**Pull request by jonesc-oci**

### CODE REVIEW

File: dds/DCPS/DiscoveryBase.h [hunk 1]
```diff
@@ -100,8 +100,8 @@ namespace OpenDDS {
         , reader_done_(false), writer_done_(false), cnd_(mtx_)
         , interval_(TimeDuration(0))
         , status_(0)
-        , tid_(0)
       {
+        tid_ = 0;
```
Comment (mitza-oci) [inline dds/DCPS/DiscoveryBase.h:104]: Why was this changed?

Comment (jonesc-oci) [reply dds/DCPS/DiscoveryBase.h:104]: The initializer caused an error in the conformance test build: DCPS/DiscoveryBase.h:103: error: no matching function for call to 'pthread_t::pthread_t(int)'

**Review by mitza-oci — APPROVED** [review]
